### DISCUSSION TITLE
Twitter OAuth 2.0

### DISCUSSION THREAD

**Discussion by camiaei** [nextauthjs/next-auth#4034] [Help] [answered]
Hi👋

I'm using the [example template](https://github.com/nextauthjs/next-auth-example
) and have enabled OAuth 2.0 following the [docs](https://next-auth.js.org/providers/twitter#oauth-2).
But when I hit the login button, I get this error:

![Screenshot (262)](https://user-images.githubusercontent.com/79164595/155007686-93162555-75e5-43ac-9719-f6807dade5b3.png)

👉Next I'm redirected to the Twitter login page, where I am asked for my username and password. Only then do I get to the page where I can authorize the app and log in.

Am I doing something wrong?😕

PS: The default OAuth 1.0 version works just fine.

**Comment by roxworks** [comment]
For anyone that ends up here - it's probably the incorrect redirect uri.

For NextAuth it's often something like /api/auth/callback/twitter

**Reply by skarthikeyan96** [reply]
Thanks @roxworks. It works. I was using `localhost:3000` as the callback

**Reply by J0SAL** [reply]
> Thanks @roxworks. It works. I was using `localhost:3000` as the callback

Hey @skarthikeyan96, how was it working for you? I tried using, http://127.0.0.1:3000/api/auth/callback/twitter and it is still not working

**Reply by reubence** [reply]
Thank you so much!

**Comment by camiaei** [comment, marked as answer]
I tried again last week and this time it worked ok.

(Make sure you use the credentials under "OAuth 2.0 Client ID and Client Secret")

**Comment by jacklynch00** [comment]
Can anyone provide any insight into how to use the users access token to make requests multiple times? I am using the next.js serverless api's to make requests after storing the users access and refresh tokens in the jwt using next-auth. I can make the first successful request, but the second request always fails and I see the following error message:

```
error: {
    error: 'invalid_request',
    error_description: 'Value passed for the token was invalid.'
}
```

**Reply by misramark** [reply]
Hello there
I'm facing the same issue here could someone help me with that
